csrc/preseg_passes/exact_mapped_extent_substitution.cpp

```diff
@@ -22,8 +22,51 @@ namespace {
 // otherwise, validateDomainEquivalence fails. If we really want to substitute,
 // we may need to skip or modify validateDomainEquivalence.
 inline bool isNonSubstitutableID(const IterDomain* id) {
-  return (id->isBroadcast() && !id->hasExpandedExtent()) || id->definition() ||
-      id->extent()->definition();
+  return id == nullptr || (id->isBroadcast() && !id->hasExpandedExtent()) ||
+      id->definition() || id->extent()->definition();
+}
+
+// Build disjoint set of extents from  disjoint set of ids
+auto buildExtentSetFromIdSets(const DisjointSets<Val*>& id_sets) {
+  DisjointSets<Val*> extent_sets;
+  // Loop over each id set
+  for (const auto& set_ptr : id_sets.disjointSets()) {
```

Comment:
Let's make the difference more explicit:
```suggestion
  for (const auto& id_set_ptr : id_sets.disjointSets()) {
```